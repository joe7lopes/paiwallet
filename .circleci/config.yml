version: 2
jobs:
  build:
    docker:
      # specify the version you desire here
      - image: circleci/node:7.10
        environment:
          NODE_ENV: production
    working_directory: ~/pai-wallet
    steps:
      - checkout  
      - run: npm install
  test:
    steps:
      - run: npm run test

workflows:
  version: 2
  build_and_test_and_deploy:
    jobs:
      - build
      - test:
          requires:
            - build
      # - deploy:
      #   requires:
      #     - test


