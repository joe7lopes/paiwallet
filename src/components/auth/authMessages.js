export const authMessages = {
    "auth/user-not-found": "User doesn't exist"
}

