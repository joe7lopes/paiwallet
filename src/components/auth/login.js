import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import {
    startGoogleLogin,
    startFacebookLogin,
    startEmailLogin,
    clearLoginError
} from '../../actions/auth_action';

class Login extends React.Component{
    constructor(props){
        super(props);
        
        this.state = {
            email: '',
            password: '',
            emailError:''
        }

        this.handleInputChange = this.handleInputChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.renderError = this.renderError.bind(this);
    }

    componentWillReceiveProps(props){
        const {error} = props
        if(error.code === 'auth/invalid-email'){
            let emailError = error.message;
            this.setState({emailError});
        }
    }

    handleInputChange(event){
        this.clearErros()
        const target = event.target;
        const value = target.value;
        const name = target.name;
        this.setState({
          [name]: value
        });
    }

    clearErros(){
        this.setState({emailError: ''});
    }

    handleSubmit(e){
        e.preventDefault();
        const { email, password } = this.state;
        this.props.startEmailLogin(email, password);
    }

    renderError(){
        const { error } = this.props;
        if (error){
            return this.getHTMLForError(error);
        }
    }

    getHTMLForError(error){
        if(error.code === 'auth/user-not-found'){
            return (
                <p className="red-text">User does not exist.
                    &nbsp;<Link to="/subscription" onClick={this.props.clearLoginError}>Create Account!</Link>
                </p>
            )
        }
    }

    render(){

        const { email, password, emailError } = this.state;
        const { startGoogleLogin, startFacebookLogin } = this.props;

        return(
            <div className="pai-container blue lighten-5" style={{color: "teal"}}>
                <form onSubmit={this.handleSubmit}>
                    <div className="input-field">
                        <i className="material-icons prefix">email</i>
                        <input 
                            id="email" 
                            name="email" 
                            type="email" 
                            required
                            className={emailError ? "invalid": "validate"} 
                            value={email} 
                            onChange={this.handleInputChange}/>
                        <label htmlFor="email" data-error={emailError}>email</label>
                    </div>
                    <div className="input-field">
                        <i className="material-icons prefix">lock_outline</i>
                        <input 
                            id="pass" 
                            name="password" 
                            type="password" 
                            className="validate"
                            required
                            value={password} 
                            onChange={this.handleInputChange}/>
                        <label htmlFor="pass" data-error="wrong">password</label>
                    </div>
                    <div className="input-field">
                        {this.renderError()}
                        <button type="submit" className="btn waves-effect waves-light fullWidth">
                            Login
                        </button>
                    </div>
                </form>
                <div className="input-field grey-text alternative-login-text">
                    Or Login With
                </div>
                <div className="row">
                    <div className="col s12 l6">
                        <a className="waves-effect waves-light btn red darken-1 fullWidth"
                            onClick={startGoogleLogin}>
                            <i className="material-icons right">add</i>
                                Google
                        </a>
                    </div>
                    <div className="col s12 l6">
                        <a className="waves-effect waves-light btn blue darken-2 fullWidth"
                            onClick={startFacebookLogin}>
                            <i className="material-icons right">face</i>
                            FaceBook
                        </a>
                    </div>
                </div>
            </div>
        );
    }
}
const mapDispatchToProps = (dispatch) => ({
    startGoogleLogin: () => dispatch(startGoogleLogin()),
    startFacebookLogin: () => dispatch(startFacebookLogin()),
    startEmailLogin: (email, password) => dispatch(startEmailLogin(email, password)),
    clearLoginError: () => dispatch(clearLoginError())
});

const mapStateToProps = (state) => {
    return {
        error: state.auth.errors
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);