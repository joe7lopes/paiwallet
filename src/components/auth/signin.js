import React from 'react';
import { Field, reduxForm } from 'redux-form';
import * as actions from '../../actions';

class Signin extends React.Component{

    handleFormSubmit({email, password}){
        console.log(email, password);
        this.props.singninUser({ email, password });
    }

    render(){

        const { handleSubmit } = this.props;

        return (
                <form onSubmit={handleSubmit(this.handleFormSubmit)}>
                    <Field name="email" type="email" component="input" placeholder="rich@gmail.com"/>
                    <Field name="password" type="password" component="input" placeholder="password"/>
                    <button type="submit">Sign in</button>
                </form>
        );
    }
}

export default reduxForm({
    form: 'signin'
})(Signin);