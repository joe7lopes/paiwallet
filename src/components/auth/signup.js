import React from 'react';

class Signup extends React.Component{
    render(){
        return(
            <div className="pai-container blue lighten-5" style={{color: "teal"}}>
                <div className="row" style={{marginBottom: '0px', marginTop: '10px'}}>
                    <form className="col s12">
                    <div className="row">
                        <div className="input-field col s12 m6">
                        <i className="material-icons prefix hide-on-med-and-down">account_circle</i>
                        <input id="first_name" type="text" className="validate"/>
                        <label htmlFor="first_name">First Name</label>
                        </div>
                        <div className="input-field col s12 m6">
                        <input id="last_name" type="text" className="validate"/>
                        <label htmlFor="last_name">Last Name</label>
                        </div>
                    </div>
                    <div className="row">
                        <div className="input-field col s12">
                        <i className="material-icons prefix hide-on-med-and-down">email</i>
                        <input id="email" type="email" className="validate"/>
                        <label htmlFor="email">Email</label>
                        </div>
                    </div>
                    <div className="row">
                        <div className="input-field col s12 m6">
                        <i className="material-icons prefix hide-on-med-and-down">locker</i>
                        <input id="pass1" type="password" className="validate"/>
                        <label htmlFor="pass1">password</label>
                        </div>
                        <div className="input-field col s12 m6">
                        <input id="pass2" type="password" className="validate"/>
                        <label htmlFor="pass2">password2</label>
                        </div>
                    </div>
                    <div className="row">
                            <div className="col m12 center">
                                <div className="input-field col m3">
                                    <button className="btn waves-effect waves-light" type="submit" name="action">Sign up</button>
                                </div>
                                <div className="input-field col m9">
                                    <button className="btn waves-effect waves-light home-login-facebook-button" type="submit" name="action">Connect with FaceBook</button>
                                </div>
                            </div>
                        </div>
                </form>
              </div>
            </div>
        );
    }
}

export default Signup;