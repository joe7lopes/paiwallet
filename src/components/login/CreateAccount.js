import React from 'react';

class CreateAccount extends React.Component {

    render() {
        return (
            <section className="hero is-light">
                <div className="hero-body">
                    <div className="container">
                        <h1 className="title">
                            Create Account
                        </h1>
                        <h2 className="subtitle">
                            Primary subtitle
                        </h2>
                    </div>
                </div>
            </section>
        );
    }
}

export default CreateAccount;