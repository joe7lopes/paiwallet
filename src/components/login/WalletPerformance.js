import React from 'react';

class WalletPerformance extends React.Component {
    render() {
        return (
            <section className="hero is-primary">
                <div className="hero-body">
                    <div className="container">
                        <h1 className="title">
                            Create Account
                        </h1>
                        <h2 className="subtitle">
                            Primary subtitle
                        </h2>
                    </div>
                </div>
            </section>
        );
    }
}

export default WalletPerformance;