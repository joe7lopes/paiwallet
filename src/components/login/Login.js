import React from 'react';
import WalletPerformance from './WalletPerformance';
import CreateAccount from './CreateAccount';

class Login extends React.Component {

    render() {
        return (
            <section className="section">
                <div className="columns">
                    <div className="column is-half">
                        <WalletPerformance/>
                    </div>
                    <div className="column is-half">
                        <CreateAccount/>
                    </div>
                </div>  
            </section>
        );
    }
}

export default Login;