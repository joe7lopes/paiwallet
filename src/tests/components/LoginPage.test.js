import React from 'react';
import {shallow} from '../config';
import { Login } from '../../components/auth/login';

describe('<Login />', () => {

    it('Should render login page', ()=>{
        const wrapper = shallow(<Login startLogin={()=>{}} />);
        expect(wrapper).toMatchSnapshot();
    });

    // it('should call startLogin on button click', () => {
    //     const startLogin = jest.fn();
    //     const wrapper = shallow(<Login startLogin={startLogin} />);
    //     wrapper.find('button').simulate('click');
    //     expect(startLogin).toHaveBeenCalled();
    //   });

});