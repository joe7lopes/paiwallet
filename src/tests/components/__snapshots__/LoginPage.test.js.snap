// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Should render login page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    startLogin={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
        >
          <div
            className="input-field"
          >
            <i
              className="material-icons prefix"
            >
              email
            </i>
            <input
              className="validate"
              id="email"
              type="text"
            />
            <label
              htmlFor="email"
            >
              email
            </label>
          </div>
          <div
            className="input-field"
          >
            <i
              className="material-icons prefix"
            >
              lock_outline
            </i>
            <input
              className="validate"
              id="pass1"
              type="password"
            />
            <label
              htmlFor="pass1"
            >
              password
            </label>
          </div>
        </form>,
        <div
          className="row"
        >
          <div
            className="col l12 center"
          >
            <div
              className="input-field col l3"
            >
              <button
                className="btn waves-effect waves-light"
                name="action"
                type="submit"
              >
                Login
              </button>
            </div>
            <div
              className="input-field col l9"
            >
              <button
                className="btn waves-effect waves-light home-login-facebook-button"
                name="action"
                onClick={[Function]}
              >
                Login with FaceBook
              </button>
            </div>
          </div>
        </div>,
      ],
      "className": "pai-container blue lighten-5",
      "style": Object {
        "color": "teal",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="input-field"
            >
              <i
                className="material-icons prefix"
              >
                email
              </i>
              <input
                className="validate"
                id="email"
                type="text"
              />
              <label
                htmlFor="email"
              >
                email
              </label>
            </div>,
            <div
              className="input-field"
            >
              <i
                className="material-icons prefix"
              >
                lock_outline
              </i>
              <input
                className="validate"
                id="pass1"
                type="password"
              />
              <label
                htmlFor="pass1"
              >
                password
              </label>
            </div>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="material-icons prefix"
                >
                  email
                </i>,
                <input
                  className="validate"
                  id="email"
                  type="text"
                />,
                <label
                  htmlFor="email"
                >
                  email
                </label>,
              ],
              "className": "input-field",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "email",
                  "className": "material-icons prefix",
                },
                "ref": null,
                "rendered": "email",
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "validate",
                  "id": "email",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "email",
                  "htmlFor": "email",
                },
                "ref": null,
                "rendered": "email",
                "type": "label",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="material-icons prefix"
                >
                  lock_outline
                </i>,
                <input
                  className="validate"
                  id="pass1"
                  type="password"
                />,
                <label
                  htmlFor="pass1"
                >
                  password
                </label>,
              ],
              "className": "input-field",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "lock_outline",
                  "className": "material-icons prefix",
                },
                "ref": null,
                "rendered": "lock_outline",
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "validate",
                  "id": "pass1",
                  "type": "password",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "password",
                  "htmlFor": "pass1",
                },
                "ref": null,
                "rendered": "password",
                "type": "label",
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col l12 center"
          >
            <div
              className="input-field col l3"
            >
              <button
                className="btn waves-effect waves-light"
                name="action"
                type="submit"
              >
                Login
              </button>
            </div>
            <div
              className="input-field col l9"
            >
              <button
                className="btn waves-effect waves-light home-login-facebook-button"
                name="action"
                onClick={[Function]}
              >
                Login with FaceBook
              </button>
            </div>
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="input-field col l3"
              >
                <button
                  className="btn waves-effect waves-light"
                  name="action"
                  type="submit"
                >
                  Login
                </button>
              </div>,
              <div
                className="input-field col l9"
              >
                <button
                  className="btn waves-effect waves-light home-login-facebook-button"
                  name="action"
                  onClick={[Function]}
                >
                  Login with FaceBook
                </button>
              </div>,
            ],
            "className": "col l12 center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn waves-effect waves-light"
                  name="action"
                  type="submit"
                >
                  Login
                </button>,
                "className": "input-field col l3",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login",
                  "className": "btn waves-effect waves-light",
                  "name": "action",
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Login",
                "type": "button",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn waves-effect waves-light home-login-facebook-button"
                  name="action"
                  onClick={[Function]}
                >
                  Login with FaceBook
                </button>,
                "className": "input-field col l9",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login with FaceBook",
                  "className": "btn waves-effect waves-light home-login-facebook-button",
                  "name": "action",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Login with FaceBook",
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
          >
            <div
              className="input-field"
            >
              <i
                className="material-icons prefix"
              >
                email
              </i>
              <input
                className="validate"
                id="email"
                type="text"
              />
              <label
                htmlFor="email"
              >
                email
              </label>
            </div>
            <div
              className="input-field"
            >
              <i
                className="material-icons prefix"
              >
                lock_outline
              </i>
              <input
                className="validate"
                id="pass1"
                type="password"
              />
              <label
                htmlFor="pass1"
              >
                password
              </label>
            </div>
          </form>,
          <div
            className="row"
          >
            <div
              className="col l12 center"
            >
              <div
                className="input-field col l3"
              >
                <button
                  className="btn waves-effect waves-light"
                  name="action"
                  type="submit"
                >
                  Login
                </button>
              </div>
              <div
                className="input-field col l9"
              >
                <button
                  className="btn waves-effect waves-light home-login-facebook-button"
                  name="action"
                  onClick={[Function]}
                >
                  Login with FaceBook
                </button>
              </div>
            </div>
          </div>,
        ],
        "className": "pai-container blue lighten-5",
        "style": Object {
          "color": "teal",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="input-field"
              >
                <i
                  className="material-icons prefix"
                >
                  email
                </i>
                <input
                  className="validate"
                  id="email"
                  type="text"
                />
                <label
                  htmlFor="email"
                >
                  email
                </label>
              </div>,
              <div
                className="input-field"
              >
                <i
                  className="material-icons prefix"
                >
                  lock_outline
                </i>
                <input
                  className="validate"
                  id="pass1"
                  type="password"
                />
                <label
                  htmlFor="pass1"
                >
                  password
                </label>
              </div>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="material-icons prefix"
                  >
                    email
                  </i>,
                  <input
                    className="validate"
                    id="email"
                    type="text"
                  />,
                  <label
                    htmlFor="email"
                  >
                    email
                  </label>,
                ],
                "className": "input-field",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "email",
                    "className": "material-icons prefix",
                  },
                  "ref": null,
                  "rendered": "email",
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "validate",
                    "id": "email",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "email",
                    "htmlFor": "email",
                  },
                  "ref": null,
                  "rendered": "email",
                  "type": "label",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="material-icons prefix"
                  >
                    lock_outline
                  </i>,
                  <input
                    className="validate"
                    id="pass1"
                    type="password"
                  />,
                  <label
                    htmlFor="pass1"
                  >
                    password
                  </label>,
                ],
                "className": "input-field",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "lock_outline",
                    "className": "material-icons prefix",
                  },
                  "ref": null,
                  "rendered": "lock_outline",
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "validate",
                    "id": "pass1",
                    "type": "password",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "password",
                    "htmlFor": "pass1",
                  },
                  "ref": null,
                  "rendered": "password",
                  "type": "label",
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col l12 center"
            >
              <div
                className="input-field col l3"
              >
                <button
                  className="btn waves-effect waves-light"
                  name="action"
                  type="submit"
                >
                  Login
                </button>
              </div>
              <div
                className="input-field col l9"
              >
                <button
                  className="btn waves-effect waves-light home-login-facebook-button"
                  name="action"
                  onClick={[Function]}
                >
                  Login with FaceBook
                </button>
              </div>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="input-field col l3"
                >
                  <button
                    className="btn waves-effect waves-light"
                    name="action"
                    type="submit"
                  >
                    Login
                  </button>
                </div>,
                <div
                  className="input-field col l9"
                >
                  <button
                    className="btn waves-effect waves-light home-login-facebook-button"
                    name="action"
                    onClick={[Function]}
                  >
                    Login with FaceBook
                  </button>
                </div>,
              ],
              "className": "col l12 center",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="btn waves-effect waves-light"
                    name="action"
                    type="submit"
                  >
                    Login
                  </button>,
                  "className": "input-field col l3",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Login",
                    "className": "btn waves-effect waves-light",
                    "name": "action",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": "button",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="btn waves-effect waves-light home-login-facebook-button"
                    name="action"
                    onClick={[Function]}
                  >
                    Login with FaceBook
                  </button>,
                  "className": "input-field col l9",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Login with FaceBook",
                    "className": "btn waves-effect waves-light home-login-facebook-button",
                    "name": "action",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Login with FaceBook",
                  "type": "button",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
